### DISCUSSION TITLE
A way to avoid zi alias?

### DISCUSSION THREAD

**Discussion by comfortablynick** [zdharma-continuum/zinit#312] [Q&A] [answered]
I have starting using `zinit` and I also use `zoxide` for directory jumping, which uses `zi` for interactive jumping. Is there a way to prevent `zinit` from taking this?

**Comment by vladdoster** [comment, marked as answer]
Ah thanks for reporting this edge case, but you can (pretty sure 😆) overcome this in a few ways.

However, to gain a bit more context, can you share how you currently load Zoxide? I don't use it personally.

---

Here are three possible solutions that I'd suggest.

### 1.

Use `z` instead of `zi`, no changes needed to how you load Zoxide.

### 2.

When calling `zoxide init`, the following flag is available:

```zsh
--cmd
```

Changes the prefix of the `z` and `zi` commands.

E.g.,

```zsh
--cmd j
```

would change the commands to (`j`, `ji`)

```zsh
--cmd cd
```

would replace the `cd` command

So you'd just need to choose different shortcut. 

### 3.

I haven't tested this, but 99% sure adding the following ice would work given how zinit loads plugins.

```zsh
atload'unalias zi'
```

And if the doesn’t work, try

```zsh
atinit'unalias zi'
```

---

Please let me know if you need anything else (i.e., clarification, solutions don't work, etc.). :^)

**Reply by vladdoster** [reply]
@comfortablynick Did any of the solutions I proposed solve your issue?

**Reply by comfortablynick** [reply]
Yes I am able to change the zoxide aliases, but my hope was to keep them the same. It would be great to be able to configure it through zinit.

**Reply by vladdoster** [reply]
The third solution is what you’d want to use then, no?

**Reply by strombringer** [reply]
This thread has been quiet for a while, but I just stumbled over the same issue in my setup.

I'm trying to get solution 3 to work (2 works, but muscle memory wants to type zi all the time), but am unsuccessful so far. I've tried both `atload` and `atinit`:
```
# Source/Load zinit
source "${ZINIT_HOME}/zinit.zsh"

# Remove "zi" alias for default zoxide alias to work
zinit ice atload'unalias zi'
```

I am using powerlevel10k instant prompt. Not sure if that influences any of the other commands. Just mentioning it because of the comment in the other thread about putting `eval "$(zoxide init zsh)"` at the end of .zshrc (which it already is for me).

I've only been using zinit for a few hours, so I'm not sure if the syntax for `zinit ice ...` is even correct 😅.

**Reply by mauromotion** [reply]
For the records, given I'm loading zsh-autosuggestions with an argument at load as my first plugin, I just had to add 'unalias zi' in the same list and it did the trick:

```bash 
# Autosuggestions
zinit wait lucid atload'_zsh_autosuggest_start; unalias zi' light-mode for \
    zsh-users/zsh-autosuggestions
```

**Comment by sm1999** [comment]
For me `z` and `zi` are taken by zoxide.
Can you test by putting `eval "$(zoxide init zsh)"` at the end of .zshrc?

**Reply by vladdoster** [reply]
Mhm, this makes me wonder if using the wait ice would delay loading of zoxide so you wouldn’t need to run the evil command.

**Comment by Al3cLee** [comment]
For the record, I use

```bash
source "$HOME/.local/share/zinit/zinit.git/zinit.zsh"

# Remove zi alias set by zinit (conflicts with zoxide's zi or other tools)
(( ${+aliases[zi]} )) && unalias zi
# The ((...)) is a condition that checks if the zi alias is there; if it is removed,
# the bare `unalias zi` could return an error like "alias zi not found"

# other configs and loadings

eval "$(zoxide init zsh)"
```
